core/src/IPlatformProxy.cs

```diff
@@ -35,6 +35,24 @@ internal interface IPlatformProxy
 
         Task<bool> IsUserLocalAsync(RequestContext requestContext);
 
+        /// <summary>
+        /// Returns the name of the calling assembly
+        /// </summary>
+        /// <returns></returns>
+        string GetCallingAssemblyName();
```

Comment:
Sorry I didn't realize this is an interface for all platforms. iOS and Android don't really have assemblies, so perhaps calling these as GetCallingAppName / GetCallingAppVersion would be better.